lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexAggregateFuture.java

```diff
@@ -64,11 +65,7 @@ public VortexAggregateFuture(final Executor executor,
   /**
    * @return the next aggregation result for the future, null if no more results.
    */
-  public synchronized AggregateResult get() throws InterruptedException {
-    if (taskletIdInputMap.isEmpty()) {
```

Comment:
I was previously using a `BlockingQueue` in a `synchronized` block with which I hold the same lock as I was putting into the queue. That was a bad Idea. This has been fixed by using a special lock for `get()`.